Units/simple-forth.d/expected.tags

```diff
@@ -0,0 +1,4 @@
+: fizz?  	input.fth	/^: fizz?  $/;"	s
```

Comment:
I think `: `should not be part of the tag name.
Am I wrong?

Reply:
probably not, you're right. But since I didn't find a way to label it "word" instead of "section", it might be safer, at the moment, to keep the ``: ``. Later it should be great to be able to add variables and constant and to have them sorted in their own subsections.

Reply:
Could you write a .b test case (https://docs.ctags.io/en/latest/testing-parser.html#gathering-test-cases-for-known-bugs) to show the ideal behavior of forth parser?

If we implement word, section, variable, and constant, what kind of input can we test the parser? I would like to see the input and imaginary tags output.

Reply:
You can have such a sample input there:
https://github.com/universal-ctags/ctags/blob/b97acf060ddb84c13ccd78d4eaea5d614db506c5/Units/simple-forth.d/input.fth

```
10 constant DUMMY1
12 constant DUMMY2

variable dummy3
variable dummy4

11 dummy3 !
13 dummy4 !

: fizz?  
3 mod 0 = dup if ." Fizz" then ;
: buzz?  
5 mod 0 = dup if ." Buzz" then ;

: fizz-buzz?  
dup fizz? swap buzz? or invert ;

: do-fizz-buzz  
25 1 do cr i fizz-buzz? if i . then loop ;
```

And the tags output could be something like:

```
10 constant DUMMY1 	input.fth	/^DUMMY1  $/;"	s
12 constant DUMMY2 	input.fth	/^DUMMY2  $/;"	s
variable dummy3 	input.fth	/^dummy3  $/;"	s
variable dummy4 	input.fth	/^dummy3  $/;"	s
```

(with w, c and v instead of s I suppose)

Reply:
to be more precise, "variable" and "constant" are also some special words in forth (in forth something is either a word or a value, there is nothing else), and therefore some forth could use CONSTANT and VARIABLE while others would use constant and variable (historically they were in capital letters). It's also possible to use either small or capital letters for the constants and variable names, but the usage tends to use capital letters for constant names and small letters for variable names, it's the reason I've written them this way in the sample